opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -298,9 +373,6 @@ def get_label_set(self, labels: Dict[str, str]):
         # Use simple encoding for now until encoding API is implemented
         encoded = tuple(sorted(labels.items()))
```

Comment:
thinking through it, there's some level of merit in not recreating the same label set over and over again, but constructing the labels dictionary is of itself quite expensive.

Reply:
Correct, I believe the optimization is so that the same label set does not need to be calculated over and over again. Currently, the encoding used is a default placeholder encoding type. There is talks about implementing a configurable encoder in the SDK, so user's can configure the type of encoding appropriate to the exporter they want to send metrics to. There is no specs for it yet but the [Go implementation](https://github.com/open-telemetry/opentelemetry-go/blob/master/sdk/export/metric/metric.go#L182) has created one. I think once we have this, vendors can pass in their own encoders with however they want to encode, so the iterate dict -> sort -> tuple algorithm does not always apply.